modules/tinymce/src/core/main/ts/api/EditorCommands.ts

```diff
@@ -52,81 +39,49 @@ export interface EditorCommandsConstructor {
-   * @param {Boolean} ui Optional user interface state.
-   * @param {Object} value Optional value for command.
-   * @param {Object} args Optional extra arguments to the execCommand.
-   * @return {Boolean} true/false if the command was found or not.
+   * @param {String} cmd Command name to execute, for example mceLink or Bold.
+   * @param {Boolean} ui True/false state if a UI (dialog) should be presented or not.
+   * @param {mixed} value Optional command value, this can be anything.
+   * @param {Object} args Optional arguments object.
+   * @return {Boolean} true or false if the command was supported or not.
    */
-  public execCommand(command: string, ui?: boolean, value?: any, args?: any): boolean {
-    let func, state = false;
-    const self = this;
+  public execCommand(command: string, ui?: boolean, value?: any, args?: ExecCommandArgs): boolean {
+    const editor = this.editor;
+    const lowerCaseCommand = command.toLowerCase();
+    const skipFocus = args?.skip_focus;
 
-    if (self.editor.removed) {
-      return;
+    if (editor.removed) {
+      return false;
     }
 
-    if (command.toLowerCase() !== 'mcefocus') {
-      if (!/^(mceAddUndoLevel|mceEndUndoLevel|mceBeginUndoLevel|mceRepaint)$/.test(command) && (!args || !args.skip_focus)) {
-        self.editor.focus();
+    if (lowerCaseCommand !== 'mcefocus') {
+      if (!/^(mceAddUndoLevel|mceEndUndoLevel)$/i.test(lowerCaseCommand) && !skipFocus) {
+        editor.focus();
       } else {
-        SelectionBookmark.restore(self.editor);
+        SelectionBookmark.restore(editor);
       }
     }
 
-    args = self.editor.fire('BeforeExecCommand', { command, ui, value });
-    if (args.isDefaultPrevented()) {
+    const eventArgs = editor.fire('BeforeExecCommand', { command, ui, value });
+    if (eventArgs.isDefaultPrevented()) {
       return false;
     }
 
-    const customCommand = command.toLowerCase();
-    if ((func = self.commands.exec[customCommand])) {
-      func(customCommand, ui, value);
-      self.editor.fire('ExecCommand', { command, ui, value });
-      return true;
-    }
-
-    // Plugin commands
-    each(this.editor.plugins, (p) => {
-      if (p.execCommand && p.execCommand(command, ui, value)) {
-        self.editor.fire('ExecCommand', { command, ui, value });
-        state = true;
-        return false;
-      }
-    });
-
-    if (state) {
-      return state;
-    }
-
-    // Theme commands
-    if (self.editor.theme && self.editor.theme.execCommand && self.editor.theme.execCommand(command, ui, value)) {
-      self.editor.fire('ExecCommand', { command, ui, value });
-      return true;
-    }
-
-    // Browser commands
-    try {
-      state = self.editor.getDoc().execCommand(command, ui, value);
-    } catch (ex) {
-      // Ignore old IE errors
-    }
-
-    if (state) {
-      self.editor.fire('ExecCommand', { command, ui, value });
+    const func = this.commands.exec[lowerCaseCommand];
+    if (func) {
```

Comment:
This might be out of scope, so feel free to ignore, but maybe add a `Type.isFunction` guard here?